src/code_80296A50.c

```diff
@@ -3466,44 +3435,29 @@ void func_8029E158(void) {
 GLOBAL_ASM("asm/non_matchings/code_actors/func_8029E158.s")
 #endif
 
-#ifdef MIPS_TO_C
-//generated by mips_to_c commit 3c3b0cede1a99430bfd3edf8d385802b94f91307
-? func_800C99E0(void *, ?, void *); // extern
+void func_8029E7DC(struct Actor *actor) {
+    s16 temp_v0 = actor->flags;
 
-void func_8029E7DC(void *arg0) {
-    s16 temp_v0;
-
-    temp_v0 = arg0->unk2;
     if ((temp_v0 & 0x200) != 0) {
-        func_800C99E0(arg0 + 0x18, 0x19019053, arg0);
+        func_800C99E0(actor->pos, 0x19019053);
         return;
     }
     if ((temp_v0 & 0x100) != 0) {
-        func_800C99E0(arg0 + 0x18, 0x19018010, arg0);
+        func_800C99E0(actor->pos, 0x19018010);
         return;
     }
     if ((temp_v0 & 0x80) != 0) {
-        func_800C99E0(arg0 + 0x18, 0x19008054, arg0);
+        func_800C99E0(actor->pos, 0x19008054);
     }
 }
-#else
-GLOBAL_ASM("asm/non_matchings/code_actors/func_8029E7DC.s")
-#endif
-
-#ifdef MIPS_TO_C
-//generated by mips_to_c commit 3c3b0cede1a99430bfd3edf8d385802b94f91307
-? func_8029E7DC(); // extern
-extern u16 gNumActors;
 
-void func_8029E854(void *arg0) {
-    func_8029E7DC();
-    arg0->unk2 = 0;
-    arg0->unk0 = 0;
-    gNumActors = gNumActors - 1;
+// Deletes an actor?
```

Comment:
pretty much, yeah. If shell or banana despawns due to getting hit this is how that actor is removed from the actor list.

Reply:
I think it's a multi step process for actor removal. This doesn't seem like enough code.